crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -184,6 +185,15 @@ impl CheckpointStore {
         Ok(self.tables.checkpoints.get(&seq)?)
     }
 
+    // TODO: there might be more efficient ways to implement this.
```

Comment:
Yeah I think one way to make this more efficient is in the CommitteeStore::committee_map, for each epoch we also store the first checkpoint sequence number along with the committee.
Feel free to take a look if you want in a follow-up PR.

Reply:
Will look into this and do a follow-up PR.